src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,132 @@
+        // If the user doesn't have any active ITFs, set expired status
+        if (activeList.length === 0) {
+          status = PRESTART_STATUSES.expired;
+          expirationDate = getLatestTimestamp(itfList.map(itf => itf.expirationDate));
+          // If the user has an active ITF, set retrieved status
+        } else {
+          status = PRESTART_STATUSES.retrieved;
+          expirationDate = activeList[0].expirationDate;
+        }
       }
-    });
+      dispatch(setPrestartStatus(status, expirationDate));
+      return status;
+    },
+    ({ errors }) => {
+      const status = hasSavedForm ? PRESTART_STATUSES.notRetrievedSaved : PRESTART_STATUSES.notRetrievedNew;
+      Raven.captureMessage('vets_itf_check_failure');
+      dispatch(setPrestartStatus(status, errors));
+      return status;
+    }
+  );
+}
 
-    return delay(2000);
-  };
+export function submitITFRequest(dispatch, successStatus, errorStatus) {
+
+  return apiRequest(
+    '/intent_to_file/compensation',
+    { method: 'POST' },
+    ({ data }) => {
+      dispatch(setPrestartStatus(successStatus, data.attributes.intentToFile.expirationDate));
+      return successStatus;
+    },
+    ({ errors }) => {
+      Raven.captureMessage('vets_itf_submission_error');
+      dispatch(setPrestartStatus(errorStatus, errors));
+      return errorStatus;
+    }
+  );
 }
 
-//   apiRequest(
-//     `${intentToFileUrl}`,
-//     null,
-//     ({ data }) => {
-//       const ITFStatus = data.attributes.ITFStatus
-//       onChange({ ITFStatus });
-//       return ITFStatus === 'active';
-//     },
-//     ({ errors }) => {
-//       const errorMessage = 'Network request failed';
-//       onChange({
-//         ITFStatus,
-//         errorMessage
-//       });
-//       Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
-//     }
-//   );
-// };
-// }
-/* eslint-enable */
+export function verifyIntentToFile(hasSavedForm) {
+  return async (dispatch) => {
+    let submitSuccessStatus; // eslint-disable-line no-unused-vars
+    let submitErrorStatus;
+    dispatch(setPrestartStatus(PRESTART_STATUSES.pending));
 
+    const existingITFStatus = await checkITFRequest(dispatch, hasSavedForm);
+
+    if (prestartFailureStatuses.has(existingITFStatus)) {
+      return false;
+    } else if (prestartSuccessStatuses.has(existingITFStatus)) {
+      return true;
+    }
+
+    if (existingITFStatus === PRESTART_STATUSES.none) {
+      submitSuccessStatus = PRESTART_STATUSES.created;
+      submitErrorStatus = PRESTART_STATUSES.notCreated;
+    } else if (existingITFStatus === PRESTART_STATUSES.expired) {
+      submitSuccessStatus = PRESTART_STATUSES.renewed;
+      submitErrorStatus = PRESTART_STATUSES.notRenewed;
+    }
+    const newITFStatus = await submitITFRequest(dispatch, submitSuccessStatus, submitErrorStatus);
+
+    if (prestartFailureStatuses.has(newITFStatus)) {
+      return false;
+    }
+    return true;
```

Comment:
So it seems that we:
1. Check ITF status
2. If an ITF needs to be submitted, we try to submit one
3. We return a boolean from `verifyIntentToFile`, whose value depends on whether we get a success or error response string back from `submitITFRequest`

What happens if `verifyIntentToFile` returns `false`? Is this taken as an indication in the front end that an error has occurred and the veteran can't proceed to the form?

Reply:
We can remove these return statements since this is no longer passed along to gating logic within `fetchInProgressForm` (which has already been run by the time this is executed). It is still useful for testing purposes? Or is the logic simple enough that we can simply take these out.

Reply:
Vestigial returns removed.